src/tests/ftest/util/soak_test_base.py

```diff
@@ -595,7 +596,11 @@ def run_soak(self, test_param):
                 self.harasser_results = {}
                 self.harassers, self.offline_harassers = get_harassers(harasser)
             if not single_test_pool and "extend-pool" in self.harassers + self.offline_harassers:
-                ranks = self.server_managers[0].get_host_ranks(self.hostlist_servers[:-1])
+                # create a pool excluding the ranks from the selected host
+                self.selected_host = self.random.choice(self.hostlist_servers)
+                selected_ranks = self.server_managers[0].get_host_ranks(self.selected_host)
+                server_ranks = self.server_managers[0].get_host_ranks(self.hostlist_servers)
+                ranks = [i for i in server_ranks if i not in selected_ranks]
```

Comment:
FYI - this should also work. Since `NodeSet() - NodeSet() -> NodeSet()`
```suggestion
                # create a pool excluding the ranks from the selected host
                self.selected_host = self.random.choice(self.hostlist_servers)
                ranks = self.server_managers[0].get_host_ranks(self.hostlist_servers - NodeSet(self.selected_host))
```